replay/models/nmslib_hnsw.py

```diff
@@ -367,34 +367,37 @@ def _infer_nmslib_hnsw_index(
         return_type = "item_idx array<int>, distance array<double>"
 
         if index_type == "sparse":
-            pandas_log = log.select(
-                "user_idx", "item_idx", "relevance"
-            ).toPandas()
-            interactions_matrix = csr_matrix(
-                (
-                    pandas_log.relevance,
-                    (pandas_log.user_idx, pandas_log.item_idx),
-                ),
-                shape=(self._user_dim, self._item_dim),
-            )
 
             if filter_seen_items:
 
                 @pandas_udf(return_type)
                 def infer_index(
                     user_idx: pd.Series,
+                    vector_items: pd.Series,
+                    vector_relevances: pd.Series,
                     num_items: pd.Series,
                     seen_item_idxs: pd.Series,
                 ) -> pd.DataFrame:
                     index_file_manager = index_file_manager_broadcast.value
-
                     index = index_file_manager.index
 
                     # max number of items to retrieve per batch
                     max_items_to_retrieve = num_items.max()
 
+                    exploded_vector_items = vector_items.explode()
```

Comment:
duplicate with the second branch of 'if'. Make a separate function that creates csr_matrix and apply in the corresponding places.